src/psycop_model_training/preprocessing/pre_split/processors/value_cleaner.py

```diff
@@ -59,6 +59,36 @@ def _negative_values_to_nan(self, dataset: pd.DataFrame) -> pd.DataFrame:
 
         return dataset
 
+    def _offset_negative_values(self, dataset: pd.DataFrame) -> pd.DataFrame:
+        """Offset values with minimum negative value, so all values will be non-negative"""
+
+        preds = dataset[infer_predictor_col_name(df=dataset)]
+
+        # Get all columns with negative values
+        cols_with_numerical_values = preds.select_dtypes(include=["number"]).columns
+
+        numerical_columns_with_negative_values = [
+            c for c in cols_with_numerical_values if preds[c].min() < 0
+        ]
+
+        df_to_replace = dataset[numerical_columns_with_negative_values].copy()
+
+        # Get minimum value in each column
+        col_min_values = [df_to_replace[c].min() for c in df_to_replace]
```

Comment:
Code based on taking two orderings and zipping them together is typically brittle - if we add anything in the future which changes the ordering, this code will zip them together without complaining, but the results will be completely wrong.

I'd propose two solutions:
1. (Mandatory): Write a test for this method. Since it doesn't use the self argument, you can convert it to a @staticmethod and just straight up apply it.
2. Convert to using a dict which maps from column name to minimum value
```suggestion
        col_min_values = {c: df_to_replace[c].min() for c in df_to_replace]}
```
And then adapt the code that comes after.

Reply:
Good point. I have changed it and added tests 👍